src/ray/raylet/local_task_manager.cc

```diff
@@ -656,10 +664,13 @@ bool LocalTaskManager::PoppedWorkerHandler(
     Dispatch(worker, leased_workers_, work->allocated_instances, task, reply, callback);
     erase_from_dispatch_queue_fn(work, scheduling_class);
     dispatched = true;
+
+    // Remove task dependencies because the task is dispatched, indicating dependencies
+    // are ready.
+    remove_task_dependencies = true;
   }
 
-  // Remove task dependencies.
-  if (!spec.GetDependencies().empty()) {
+  if (remove_task_dependencies && !spec.GetDependencies().empty()) {
```

Comment:
Basically we want to make sure we call `RemoveTaskDependencies` when the task is removed from the dispatch queue (the corresponding `RequestTaskDependencies` is called when the task is added to the dispatch queue).

So I think we can just move the call into `erase_from_dispatch_queue_fn`?

Reply:
done